bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -18,47 +18,624 @@ package v1alpha1
+	// +optional
+	NodeNamePrefix string `json:"nodeNamePrefix,omitempty"`
+
+	// NodeLabel Registering and starting kubelet with set of labels
+	// +optional
+	NodeLabels []string `json:"nodeLabel,omitempty"`
+
+	// NodeTaint Registering kubelet with set of taints
+	// +optional
+	NodeTaints []string `json:"nodeTaint,omitempty"`
+
+	// TODO: Remove ImageCredentialProviderBinDir when implementation is understood.
+	// ImageCredentialProviderBinDir The path to the directory where credential provider plugin binaries are located (default: "/var/lib/rancher/credentialprovider/bin")
+	// +optional
+	//ImageCredentialProviderBinDir string `json:"imageCredentialProviderBinDir,omitempty"`
+
+	// ImageCredentialProviderConfigMap is a reference to the ConfigMap that contains credential provider plugin config
+	// The configMap should contain a YAML file content + a Path to the Binaries for Credential Provider.
+	// +optional
+	ImageCredentialProviderConfigMap ConfigMapRef `json:"imageCredentialProviderConfigMap,omitempty"`
+
+	// TODO: Remove ContainerRuntimeEndpoint since this feature will probably not be offered by CAPI Bootstrap provider?
+
+	// ContainerRuntimeEndpoint Disable embedded containerd and use alternative CRI implementation
+	// +optional
+	ContainerRuntimeEndpoint string `json:"containerRuntimeEndpoint,omitempty"`
+
+	// Snapshotter Override default containerd snapshotter (default: "overlayfs")
+	// +optional
+	Snapshotter string `json:"snapshotter,omitempty"`
+
+	// TODO: Find a way to handle IP addresses that should be advertised but that RKE2 cannot find on the host (Example: Elastic IPs on Cloud Providers)
+
+	// NodeIp IPv4/IPv6 addresses to advertise for node
+	// +optional
+	//NodeIp string `json:"nodeIp,omitempty"`
+
+	// NodeExternalIp IPv4/IPv6 external IP addresses to advertise for node
+	// +optional
+	// NodeExternalIp string `json:"nodeExternalIp,omitempty"`
+
+	// ResolvConf Kubelet resolv.conf file
+	// +optional
+	ResolvConf string `json:"resolvConf,omitempty"`
+
+	// ProtectKernelDefaults Kernel tuning behavior. If set, error if kernel tunables are different than kubelet defaults.
+	// +optional
+	ProtectKernelDefaults string `json:"protectKernelDefaults,omitempty"`
+
+	// SystemDefaultRegistry Private registry to be used for all system images
+	// +optional
+	SystemDefaultRegistry string `json:"systemDefaultRegistry,omitempty"`
+
+	// Selinux Enable SELinux in containerd
+	// +optional
+	Selinux string `json:"selinux,omitempty"`
+
+	// LbServerPort Local port for supervisor client load-balancer. If the supervisor and apiserver are not colocated an additional port 1 less than this port will also be used for the apiserver client load-balancer. (default: 6444)
+	// +optional
+	LbServerPort string `json:"lbServerPort,omitempty"`
+
+	// Cni CNI Plugins to deploy, one of none, calico, canal, cilium; optionally with multus as the first value to enable the multus meta-plugin (default: canal)
```

Comment:
we can make this field an enum

Reply:
If you are mentioning the Cni attribute, it is an enum with Kubebuilder validation on type (not on field). Is there a reason I should do it differently ?